packages/compiler/src/server/completion.ts

```diff
@@ -404,6 +404,19 @@ function addIdentifierCompletion(
   { program, completions }: CompletionContext,
   node: IdentifierNode,
 ) {
+  // Process the three keyword hints of the meta properties separately, such as `::type`, `::parameters`, `::returnType`
+  if (node.parent && "selector" in node.parent && node.parent.selector === "::") {
+    ["type", "parameters", "returnType"].map((label) => {
```

Comment:
i think you can check its base's type, only show "type" for model, and "parameters", "returnType" for operation